pkg/utils/environment/cpu_analyzing.go

```diff
@@ -0,0 +1,56 @@
+package environment
```

Comment:
This entire file can be replaced by [sysconf implementation](https://pkg.go.dev/github.com/tklauser/go-sysconf) for golang OR something as simple as:

```go
// #cgo LDFLAGS:
// #include <unistd.h>
import "C"
import "fmt"
func main() {
    fmt.Printf("%d\n", C.sysconf(C._SC_NPROCESSORS_CONF))
}
```

Reply:
Nice!
Thanks for this comment, I didn't know it was implemented so easily to use.
We need to tell Datadog ;)

Reply:
Please don't do that!

Also see this thread:
https://lore.kernel.org/bpf/CAEf4Bza6Bsee1i_ypbDogG5MsVFGW9pnatxHCn9PycW9eP2Gkw@mail.gmail.com/

Reply:
Hum, okay, I've tested in multiple VMs attaching and detaching CPUs before suggesting it but, likely, I haven't tested it enough. Thanks @yanivagman

Reply:
@AlonZivony could you add a comment with that link then ? So in the future we always remember why that logic exists instead ?

Reply:
No problem